src/Microsoft.OData.Core/Json/ODataUtf8JsonWriter.Stream.cs

```diff
@@ -0,0 +1,375 @@
+                    // to the current chunk being processed.
+                    if (this.buffer != null)
+                    {
+                        // Get unprocessed bytes from the buffer.
+                        ReadOnlySpan<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsSpan().Slice(bufferPosition - bytesNotWrittenFromPreviousChunk, bytesNotWrittenFromPreviousChunk);
+                        int totalLength = bytesNotProcessedFromPreviousChunk.Length + chunk.Length;
+
+                        // Rent an array to hold bytes from both previous and current chunks.
+                        byte[] combinedArray = ArrayPool<byte>.Shared.Rent(totalLength);
+
+                        // Copy bytes from bytesNotProcessedFromPreviousChunk to the combined array
+                        bytesNotProcessedFromPreviousChunk.CopyTo(combinedArray);
+
+                        // Copy bytes from chunk to the combined array, starting from the end of bytesNotProcessedFromPreviousChunk
+                        chunk.CopyTo(combinedArray.AsSpan().Slice(bytesNotProcessedFromPreviousChunk.Length));
+
+                        WriteChunk(combinedArray,totalLength, isFinalBlock);
+
+                        if (combinedArray != null)
+                        {
+                            ArrayPool<byte>.Shared.Return(combinedArray);
+                        }
+                    }
+                    else
+                    {
+                        WriteChunk(chunk,chunk.Length, isFinalBlock);
+                    }
+
+                    // Flush the writer if the buffer threshold is reached
+                    this.jsonWriter.FlushIfBufferThresholdReached();
+                }
+            }
+
+            /// <summary>
+            /// Writes the byte value represented by the provided read-only memory in chunks using Base64 encoding.
+            /// </summary>
+            /// <param name="value">The read-only memory containing the byte value to be written.</param>
+            private async ValueTask StartWritingByteValueInChunksAsync(ReadOnlyMemory<byte> value)
+            {
+                // Process the input value in chunks
+                for (int i = 0; i < value.Length; i += chunkSize)
+                {
+                    int remainingBytes = Math.Min(chunkSize, value.Length - i);
+                    bool isFinalBlock = false ;
+
+                    // Take a chunk of bytes from the input value.
+                    ReadOnlyMemory<byte> chunk = value.Slice(i, remainingBytes);
+
+                    // If the buffer is not empty, then we copy the bytes from the buffer
+                    // to the current chunk being processed.
+                    if (this.buffer != null)
+                    {
+                        ReadOnlyMemory<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsMemory().Slice(bufferPosition - bytesNotWrittenFromPreviousChunk, bytesNotWrittenFromPreviousChunk);

```

Comment:
I find the naming of `bytesNotProcessedFromPreviousChunk` and `bytesNotWrittenFromPreviousChunk` confusing. The two names mean pretty much the same thing but refer to different values of different types. I would propose renaming at least one of them
```suggestion
                        ReadOnlyMemory<byte> bytesNotProcessedFromPreviousChunk = this.buffer.AsMemory().Slice(bufferPosition - this.numBytesNotWrittenFromPreviousChunk, this.numBytesNotWrittenFromPreviousChunk);

```